test/system/src/test/java/io/pravega/test/system/ReaderCheckpointWithInternalThreadpool.java

```diff
@@ -0,0 +1,239 @@
+        streamManager = StreamManager.create(clientConfig);
+
+        connectionFactory = new SocketConnectionFactoryImpl(clientConfig);
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(clientConfig)
+                .build(), connectionFactory.getInternalExecutor());
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        controller.close();
+        connectionFactory.close();
+    }
+
+    /*This test case is to call initaite checkpoint API with internal excutor thread pool.
+    * Number of thread to construct SocketConnectionFactory should be default to more than 1 in ReaderGroupManager constructor
+    * with single thread we do see that when readers read continuously and in parallel if we get request for initiate checkpoint then that checkpoint never completes
+     */
+
+    @Test
+    public void initiateCheckpointTest() {
+        assertTrue("Creating Scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM, streamConfig));
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, clientConfig);
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(SCOPE, STREAM))
+                .disableAutomaticCheckpoints()
+                .build();
+        readerGroupManager.createReaderGroup(READER_GROUP_NAME, readerGroupConfig);
+
+        @Cleanup
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(READER_GROUP_NAME);
+
+        // Write events
+        int start = 1;
+        int end = 500;
+        log.info("Write events with range [{},{})", start, end);
+        writeEventsToStream(SCOPE, IntStream.range(start, end).boxed().collect(Collectors.toList()));
+        asyncReadEvents(SCOPE, READER);
+        //Initiate 2 checkpoints and make sure its completed
+        initiateCheckpoointAndVerify(readerGroup);
+    }
+
+    private Checkpoint initiateCheckpoint(final ReaderGroup readerGroup, final String checkPointName) {
+        Checkpoint result = null;
+        log.info("Called Initiating checkpoint for {} ", checkPointName);
+        CompletableFuture<Checkpoint> cp = readerGroup.initiateCheckpoint(checkPointName);
+
+        try {
+            result = cp.get(5, TimeUnit.SECONDS);
+            log.info("Checkpoint isDone {} for checkpoint {} ", cp.isDone(), result.getName());
+        } catch (Exception e)  {
+            if (e instanceof InterruptedException || e instanceof ExecutionException || e instanceof TimeoutException) {
+                log.error("Exception while initiating checkpoint: {}", checkPointName, e);
+                fail("Exception while initiating checkpoint is not expected");
+            }
+        }
+        return result;
+    }
+
+    private void initiateCheckpoointAndVerify(final ReaderGroup readerGroup) {
```

Comment:
There is one typo, Could you please change the method name to `initiateCheckpointAndVerify`?

Reply:
Done